crates/primitives-traits/src/transaction/mod.rs

```diff
@@ -1,28 +1,136 @@
-//! Transaction abstraction
+use core::{fmt::Debug, hash::Hash};
 
-pub mod signed;
-
-use alloc::fmt;
+use alloy_primitives::{Bytes, TxKind, B256, U256};
+use alloy_serde::WithOtherFields;
 
+use alloy_consensus::{TxEip1559, TxEip2930, TxEip4844, TxEip7702, TxLegacy};
+use alloy_rlp::Encodable;
 use reth_codecs::Compact;
 use serde::{Deserialize, Serialize};
 
-/// Helper trait that unifies all behaviour required by transaction to support full node operations.
-pub trait FullTransaction: Transaction + Compact {}
-
-impl<T> FullTransaction for T where T: Transaction + Compact {}
+pub mod signed;
 
-/// Abstraction of a transaction.
-pub trait Transaction:
-    alloy_consensus::Transaction
+/// Inner trait for a raw transaction.
+///
+/// Transaction types were introduced in [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718).
+pub trait TransactionInner<T>:
```

Comment:
it should work to make this private. it's better because the `Transaction` trait will be imported instead, and never `TransactionInner`, since otherwise that would cause bug if arbitrary feature is set

Reply:
That was what I was thinking as well but wouldn't this mean that someone that wants to implement the `Transaction` trait will never be able to because `TransactionInner` is private?

Reply:
then let's just rename the `TransactionInner` trait to `Transaction`, and have the `FullTransaction` trait twice, once abritrary feature gated, and once not.